tonic-reflection/src/server.rs

```diff
@@ -0,0 +1,323 @@
+use crate::proto::server_reflection_request::MessageRequest;
+use crate::proto::server_reflection_response::MessageResponse;
+use crate::proto::server_reflection_server::{ServerReflection, ServerReflectionServer};
+use crate::proto::{
+    FileDescriptorResponse, ListServiceResponse, ServerReflectionRequest, ServerReflectionResponse,
+    ServiceResponse,
+};
+use prost::Message;
+use prost_types::{
+    DescriptorProto, EnumDescriptorProto, FieldDescriptorProto, FileDescriptorProto,
+};
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use thiserror::Error;
+use tokio::stream::StreamExt;
+use tokio::sync::mpsc;
+use tonic::{Request, Response, Status, Streaming};
+
+/// Represents an error in the construction of a gRPC Reflection Service.
+#[derive(Debug, Error)]
```

Comment:
we could probably do this by hand tbh, I'd rather not pull thisiserror into the stack.

Reply:
That's one of the points in the PR notes above - `thiserror` is just a temporary convenience thing until I know all the variants to implement it by hand in one go. I'll remove it before converting this into a PR.

Reply:
oops sorry! missed that